sklearn/metrics/tests/test_classification.py

```diff
@@ -148,6 +146,15 @@ def test_precision_recall_f1_score_binary():
                                       **kwargs),
                             (1 + 2 ** 2) * ps * rs / (2 ** 2 * ps + rs), 2)
 
+    # Make sure average='micro' raises ValueError
+    assert_raises_verror = partial(assert_raise_message, ValueError,
```

Comment:
I don't see why this should be banned. Certainly, I've seen macro-averages reported in binary problems, whether right or wrong from a methodological perspective.